### PR TITLE
2210 The "no stock available" message needs to be set for prescriptions

### LINKED ISSUE #2210: The "no stock available" message needs to be set for prescriptions
## What went wrong? 😲
 
When I try to dispense stock and have no stock available, it gives me an error but it's not in user-friendly language. 
It's a different behaviour to the outbound shipment which doesn't give me an error. 

## Expected behaviour 🤔
I would like a user friendly message! 

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Go to prescriptions
2. Click on 'Add item'
3. Try to add an item with no stock

Prescription behaviour:  
image

Outbound shipment behaviour: 
image

## Your environment 🌱

 - Version: 1.3.0
 - Platform:
   - [ ] android (tablet)
   - [x] browser Chrome :) Both on demo server and my local 
   - [ ] desktop (windows)
   - [ ] desktop (macOS)
   - [ ] server (windows)

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #2210 

# 👩🏻‍💻 What does this PR do? 
Display no stock available message for prescription and outbound shipments (see screenshots below). Have also hid the prescriptions note component if there is no stock line for that item.
#### Prescription
Screenshot 2023-09-12 at 8 14 35 AM
#### Outbound
Screenshot 2023-09-12 at 8 17 15 AM

# 🧪 How has/should this change been tested? 
- [ ] Go to outbound/prescription
- [ ] Try add item with no stock line
- [ ] See message

### CODE REVIEW

File: client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/hooks/useDraftOutboundLines.tsx [hunk 1]
```diff
@@ -32,11 +32,12 @@ export const useDraftOutboundLines = (
   const [draftStockOutLines, setDraftStockOutLines] = useState<
     DraftStockOutLine[]
   >([]);
+  const noStockLines = !data?.nodes.length;
 
   useConfirmOnLeaving(isDirty);
 
   useEffect(() => {
-    if (!item) {
+    if (!item || noStockLines) {
```
Comment (CarlosNZ) [inline client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/hooks/useDraftOutboundLines.tsx:40]: Just wondering if this is necessary. It still works fine without this check, as the subsequent `rows.map...` has nothing in it so ends up basically the same.

Was there a particular reason for adding this check?

**Review by CarlosNZ — APPROVED** [review]
Looks good, thanks. Just one little nitpick question in the comments -- it doesn't seem to change anything that impacts this PR specifically.

File: client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/hooks/useDraftOutboundLines.tsx (hunk 1 above)
Comment (roxy-dao) [reply client/packages/invoices/src/OutboundShipment/DetailView/OutboundLineEdit/hooks/useDraftOutboundLines.tsx:40]: It adds an empty stockOutLine to the array, so the messages don't show up...

**Review by mark-prins — APPROVED** [review]
👍  very nice

It made me wonder about disabling the `OK` and `OK & Next` buttons - there's another discussion to have around that.. the problem of users entering a quantity in, forgetting to `Allocate` and clicking `OK`. that's for another day

**Pull request merged**